demo/procedural_generator/procedural_generator.gd

```diff
@@ -93,11 +95,27 @@ func add_next_rp(rp: RoadPoint, dir: int) -> void:
 	# Copy initial things like lane counts and orientation
 	new_rp.copy_settings_from(rp, true)
 
+	new_rp.traffic_dir=[]
+	new_rp.lanes=[]
+
+	randomize()
+	for i in range(randi()%4 + 1):
+		new_rp.traffic_dir.append(LaneDir.REVERSE)
+		new_rp.lanes.append(LaneType.SLOW)
+	for i in range(randi()%3):
+		new_rp.traffic_dir.append(LaneDir.REVERSE)
+		new_rp.lanes.append(LaneType.FAST)
+	for i in range(randi()%3):
+		new_rp.traffic_dir.append(LaneDir.FORWARD)
+		new_rp.lanes.append(LaneType.FAST)
+	for i in range(randi()%4 + 1):
+		new_rp.traffic_dir.append(LaneDir.FORWARD)
+		new_rp.lanes.append(LaneType.SLOW)
```

Comment:
I usually would just use the RoadPoint class which is globally available, and not prone to path change issues. Since we only reference these on these lines, I'd suggest the below: 


```suggestion
		new_rp.traffic_dir.append(RoadPoint.LaneDir.REVERSE)
		new_rp.lanes.append(RoadPoint.LaneType.SLOW)
	for i in range(randi()%3):
		new_rp.traffic_dir.append(RoadPoint.LaneDir.REVERSE)
		new_rp.lanes.append(RoadPoint.LaneType.FAST)
	for i in range(randi()%3):
		new_rp.traffic_dir.append(RoadPoint.LaneDir.FORWARD)
		new_rp.lanes.append(RoadPoint.LaneType.FAST)
	for i in range(randi()%4 + 1):
		new_rp.traffic_dir.append(RoadPoint.LaneDir.FORWARD)
		new_rp.lanes.append(RoadPoint.LaneType.SLOW)
```